table/column.go

```diff
@@ -371,7 +373,35 @@ func GetColOriginDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (ty
 
 // GetColDefaultValue gets default value of the column.
 func GetColDefaultValue(ctx sessionctx.Context, col *model.ColumnInfo) (types.Datum, error) {
-	return getColDefaultValue(ctx, col, col.GetDefaultValue())
+	defaultValue := col.GetDefaultValue()
+	if !col.DefaultIsExpr {
+		return getColDefaultValue(ctx, col, defaultValue)
+	}
+	return getColDefaultExprValue(ctx, col, defaultValue)
+}
+
+func getColDefaultExprValue(ctx sessionctx.Context, col *model.ColumnInfo, defaultValue interface{}) (types.Datum, error) {
+	x, ok := defaultValue.(string)
+	if !ok {
+		// TODO: refine the error.
+		return types.Datum{}, errors.New("can't cast default value as expression string")
```

Comment:
Please try to refine the error, we should try to avoid errors with the unknown code.